Code/Robot/Robot/Robot.ino

```diff
@@ -21,28 +21,56 @@ int main() {
 }
 
 void loop() {
-  bool start;
+  bool start_signal;
   states state = DRIVING;
+  String inputString = "";            // Holds incoming data from Serial3
+  boolean messageInProgress = false;  // Indicates if message parsing is in progress
   while (true) {
-    // Check if data is available to read from Serial3
-    if (Serial3.available() > 0) {  // zero will change here so I'm not going to try and predict what it will be
-      // Read the incoming byte from Serial3
-      char incomingByte = Serial3.read();
-
-      // Check if the incoming byte is '1' or '0' and set the start flag accordingly
-      if (incomingByte == '1') {
-        start = true;
-      } else if (incomingByte == '0') {
-        start = false;
-      }
+
+    String message = read_serial(inputString, messageInProgress);
+    
+    // Check if the incoming byte is '1' or '0' and set the start flag accordingly
+    if (message == "") {
+
+    } else if (message == "<1>") {
```

Comment:
### **One thing I think we should change:**

'Waiting to start' can be a state instead of using the start signal bool. The 'waiting to start' state will be when our robot is at home. 

While in this state, the robot should continuously check for serial messages. If a message is received, and that message tells the robot to start, then it should transition to the 'driving' state.